src/tiledquickplugin/maploader.cpp

```diff
@@ -46,7 +46,9 @@ void MapLoader::setSource(const QUrl &source)
 
     Tiled::MapReader mapReader;
 
-    Tiled::Map *map = mapReader.readMap(source.toLocalFile());
+    const QString qrcScheme = QStringLiteral("qrc");
+    const bool isQrcScheme = source.scheme() == qrcScheme;
+    Tiled::Map *map = mapReader.readMap(isQrcScheme ? source.toString().mid(qrcScheme.length()) : source.toLocalFile());
```

Comment:
My experience has been that you generally only use the qrc scheme in QML - when it comes to C++, it's always ":/my/path". That's probably a simplified understanding of the problem, but it is supported by e.g. [this code](url). If you think the way I've done it is weird, I could copy that code.